include/library.h

```diff
@@ -146,6 +147,7 @@ class Library
 {
   std::map<std::string, kiwix::Book> m_books;
   std::map<std::string, std::shared_ptr<Reader>> m_readers;
+  std::map<std::string, std::shared_ptr<zim::Archive>> m_archives;
```

Comment:
@mgautierfr Even though all the tests are passing, the addition of this line causes a segmentation fault when I try to run `kiwix-serve` :confused: Can you help figure it out? I can confirm the issue is isolated to this one line irrespective of other changes.

Reply:
@maneeshpm Would be good to extend the tests, so the CI fails as well.

Reply:
@kelson42 Good idea! Let me try that, maybe it will reveal the underlying issue.

Reply:
@kelson42 I got this issue because, for some reason, I had to rebuild kiwix-tools again against the modified libkiwix even though no API has been changed. I guess it's an isolated incident with my system. Closing this discussion.

Reply:
good!

Reply:
This is what we call a ABI break.
If you add a member to a structure, the size of this structure is changed.
So all the code of kiwix-tools compiled with previous version is using the old size of library but the new library is using the new size. So the new libkiwix writes data where it should not (where kiwix-tools has not reserved space for it).

Reply:
Aaah cool. The segmentation fault makes sense now, Thanks!